packages/account/src/Components/forms/personal-details-form.jsx

```diff
@@ -151,27 +158,29 @@ const PersonalDetailsForm = props => {
                             />
                         )}
                         {'salutation' in values && (
-                            <RadioGroup
-                                className='dc-radio__input'
-                                name='salutation'
-                                selected={values.salutation}
-                                onToggle={e => {
-                                    e.persist();
-                                    setFieldValue('salutation', e.target.value);
-                                }}
-                                required
-                            >
-                                {salutation_list.map(item => (
-                                    <RadioGroup.Item
-                                        key={item.value}
-                                        label={item.label}
-                                        value={item.value}
-                                        disabled={
-                                            !!values.salutation && isFieldImmutable('salutation', editable_fields)
-                                        }
-                                    />
-                                ))}
-                            </RadioGroup>
+                            <span onClick={handleSalutationSelection}>
+                                <RadioGroup
+                                    className='dc-radio__input'
+                                    name='salutation'
+                                    selected={selected_salutation}
+                                    onToggle={e => {
+                                        e.persist();
+                                        setFieldValue('salutation', e.target.value);
+                                    }}
+                                    required
+                                >
+                                    {salutation_list.map(item => (
+                                        <RadioGroup.Item
+                                            key={item.value}
```

Comment:
Seems like this is not in the scope of your PR, if you can, would you please change the `key` to be something different than `value` as it's not unique to be exact in terms of radio button, to maybe `label`